### PR TITLE
kubectl explain: use openapiv3 by default

### PR DESCRIPTION
**Pull request by alexzielenski**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Switches kubectl explain default from buggy openapiv2 data to using new richer v3 data. The change in output should be strictly better/richer.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Switched kubectl explain to use OpenAPIV3 information published by the server. OpenAPIV2 backend can  still be used with the `--output plaintext-openapiv2` argument
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP] 3515 OpenAPI v3 for kubectl explain: https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/3515-kubectl-explain-openapiv3
```

/sig cli
/assign @apelisse

Depends on #115480 to be merged first

### CODE REVIEW

**Comment by apelisse** [comment]
LGTM for the last of the 4 commits, thanks.

**Comment by rjsadow** [comment]
/release-note-edit

```release-note Switched kubectl explain to use OpenAPIV3 information published by the server. OpenAPIV2 backend can  still be used with the `--output plaintext-openapiv2` argument ```

File: staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go [hunk 1]
```diff
@@ -82,10 +78,9 @@ type ExplainOptions struct {
 
 func NewExplainOptions(parent string, streams genericclioptions.IOStreams) *ExplainOptions {
 	return &ExplainOptions{
-		IOStreams:       streams,
-		CmdParent:       parent,
-		EnableOpenAPIV3: cmdutil.ExplainOpenapiV3.IsEnabled(),
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go:87]: Is it time to also remove cmdutil.ExplainOpenapiV3?

**Comment by ardaguclu** [comment]
Is this PR also fixing this issue https://github.com/kubernetes/kubernetes/issues/113902?

**Comment by apelisse** [comment]
Thanks
/approve

File: staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go (hunk 1 above)
Comment (alexzielenski) [reply staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go:87]: yes prob

**Pull request merged**